### PR TITLE
Activate the Web Application Firewall

### LINKED ISSUE #3795: Exclude Akamai requests in WAF
We're seeing a lot of false positive requests to our WAF, as Akamai has a nasty habit of not sending a user agent string. We should disable the specific firewall rule causing this concern.

In addition, some basic requests are being flagged for a missing "accept" header. Let's prune these.

Group: REQUEST-920-PROTOCOL-ENFORCEMENT
Rule ID: 920320, 920300

### PR DESCRIPTION
**Pull request by rin-skylight**
# DEVOPS PULL REQUEST

## Related Issue

- Resolves #3795.
- Completes generation of WAF exceptions

## Changes Proposed

- Add remaining argument exceptions that are known to cause problems
- Activate the WAF in **Prevention** mode. 

## Additional Information

- Support will need to be briefed regarding this change before it goes live, as it may cause failed requests in very unique scenarios for WAF elements that we have not had the opportunity to refine.
- This will likely require continued tuning over time. As the application changes, we will see additional scenarios that may need to be excepted.
- This will go live on ALL environments with this change. Developers, please ensure your new features are tested on live environments. Let DevOps know if requests are being blocked, or are not behaving as expected, and we can help you add an exception to your feature PR.
 
### Infrastructure
- [ ] Consult the results of the `terraform-plan` job inside the "Terraform Checks" workflow run for this PR. Confirm that there are no unexpected changes!

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

## Cloud
- [ ] Oncall has been notified if this change is going in after-hours
- [ ] If there are changes that cannot be tested locally, this has been deployed to our Azure `test`, `dev`, or `pentest` environment for verification

### CODE REVIEW

**Review by emyl3 — COMMENTED** [review]
Just one question for you, Rin! 😸

File: ops/stg/main.tf [hunk 1]
```diff
@@ -97,6 +97,7 @@ module "app_gateway" {
     module.simple_report_api.app_hostname
   ]
 
+  firewall_policy_id = module.web_application_firewall.web_application_firewall_id
```
[the remark this answers is not in the document]
Comment (rin-skylight) [reply ops/stg/main.tf:100]: That's a great question! `Prod` already has this in place (`Dev`, as well), as part of the initial testing effort for this. This PR gets the WAF set up on all of our remaining environments.

[the remark this answers is not in the document]
Comment (emyl3) [reply ops/stg/main.tf:100]: Got it! Thank you!!

**Review by emyl3 — DISMISSED** [review]
Looks good to me. Thank you for all the hard work on this 🧙

**Review by alismx — APPROVED** [review]
Thanks for your continued work on the WAF!

**Comment by rin-skylight** [comment]
Per our discussion at today's Eng Sync, the alerting changes have now been added to this PR.

File: ops/demo/main.tf [hunk 2]
```diff
@@ -113,15 +113,15 @@ module "nat_gateway" {
...
+module "web_application_firewall" {
   source                  = "../services/web_application_firewall"
   name                    = local.name
   env                     = local.env
   resource_group_location = data.azurerm_resource_group.rg.location
   resource_group_name     = data.azurerm_resource_group.rg.name
 
   tags = local.management_tags
-} */
+}
```
Comment (nathancrtr) [inline ops/demo/main.tf:124]: Does `firewall_policy_id` need to be set here too? You [mentioned](https://github.com/CDCgov/prime-simplereport/pull/3919/files#r911367962) it's already present on `dev` and `prod`, not sure about `demo` though.

Comment (rin-skylight) [reply ops/demo/main.tf:125]: Excellent catch, thank you for this! This was just fixed.

**Review by mehansen — APPROVED** [review]

**Pull request merged**